app/models.py

```diff
@@ -0,0 +1,58 @@
+from werkzeug.security import generate_password_hash, check_password_hash
+from datetime import datetime
+
+rsvps = db.Table('rsvps',
+	db.Column('user_id', db.Integer, db.ForeignKey('users.id')),
+	db.Column('event_id', db.Integer, db.ForeignKey('events.id')),
+	)
+
+class User(db.Model):
+	"""this class to defines the user data model"""
+	__tablename__ = 'users'
+	id = db.Column(db.Integer, primary_key=True)
+	username = db.Column(db.String(64), unique=True)
+	email = db.Column(db.String(64), unique=True)
+	password_hash = db.Column(db.String(128))
+	events = db.relationship('Events', backref='created_by')
+
+	@property
+	def password(self):
```

Comment:
what does this function do? is it used in the app?

Reply:
it prevents reading the password field. bellow i am also defining a password setter.So  if you are interacting with an instance of the model you can set the password by `user.password =1234' but you if you do `user.password` it raises the `AttributeError`